### PR TITLE
Fix/order hook logic

### PR DESCRIPTION
**Pull request by Alexangelj**
Opening pr so changes can be looked at, not finished!

### CODE REVIEW

**Comment by Alexangelj** [comment]
@zachdt 
itsover

**Comment by zachdt** [comment]
I've been keeping track, but out of solidarity I'll review this in full again.

**Comment by Alexangelj** [comment]
This is the minimum work for this, but every order type is working 100% of the time and correctly. This new refactor also lets us make improvements elsewhere

**Review by zachdt — APPROVED** [review]
Legendary.  Through this tedium, you have ushered a golden era of primitive-sdk extendability.  LGTM

**Comment by zachdt** [comment]
No conflicts?  Wow.

File: app/src/lib/entities/market.ts [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    option: Option,
+    tokenAmountA: TokenAmount,
+    tokenAmountB: TokenAmount
+  ) {
+    super(tokenAmountA, tokenAmountB)
+    this.option = option
+  }
+
+  public get FEE(): BigNumberish {
+    return 301
```
[the remark this answers is not in the document]
Comment (Alexangelj) [reply app/src/lib/entities/market.ts:20]: yeah remember this 
image

**Pull request merged**